erpnext/crm/doctype/appointment/appointment.py

```diff
@@ -104,16 +104,28 @@ def create_lead_and_link(self):
 		# Return if already linked
 		if self.party:
 			return
+
 		lead = frappe.get_doc(
 			{
 				"doctype": "Lead",
 				"lead_name": self.customer_name,
 				"email_id": self.customer_email,
-				"notes": self.customer_details,
 				"phone": self.customer_phone_number,
 			}
 		)
+
+		if self.customer_details:
+			lead.append(
+				"notes",
+				{
+					"note": self.customer_details,
+					"added_by": frappe.session.user,
```

Comment:
Ideally, this should be `self.customer_name`, but I think that may cause the avatar to not load in the notes UI.